File: tools/schema/generator/gotemplates/funcs.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package gotemplates

var funcsGo = map[string]string{
	// *******************************
	"funcs.go": `
package $package$+impl

import (
	"$module/go/$package$+impl"
	"github.com/iotaledger/wasp/packages/wasmvm/wasmlib/go/wasmlib"
	"github.com/iotaledger/wasp/packages/wasmvm/wasmlib/go/wasmlib/wasmtypes"
)
$#each func funcSignature
`,
	// *******************************
	"funcSignature": `

func $kind$FuncName(ctx wasmlib.Sc$Kind$+Context, f *$FuncName$+Context) {
$#emit init$Kind$FuncName
}
`,
	// *******************************
	"initFuncInit": `
	if f.Params.Owner().Exists() {
		f.State.Owner().SetValue(f.Params.Owner().Value())
		return
	}
	f.State.Owner().SetValue(ctx.RequestSender())
`,
	// *******************************
	"initFuncSetOwner": `
	f.State.Owner().SetValue(f.Params.Owner().Value())
`,
	// *******************************
	"initViewGetOwner": `
	f.Results.Owner().SetValue(f.State.Owner().Value())
`,
}

File: contracts/wasm/gascalibration/executiontime/ts/executiontimeimpl/funcs.ts
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import * as wasmlib from "wasmlib";
import * as wasmtypes from "wasmlib/wasmtypes";
import * as sc from "../executiontime/index";

export function funcF(_ctx: wasmlib.ScFuncContext, f: sc.FContext): void {
    const n = f.params.n().value();
    let x: u32 = 0;
    let y: u32 = 0;

    for (let i: u32 = 0; i < n; i++) {
        x += 1;
        y = 3 * (x % 10);
    }
    f.results.n().setValue(y);
}
